istio-operator/internal/controller/istio/cert/cert_manager.go

```diff
@@ -40,6 +40,11 @@ func (cm *CertManager) TryCreate(ctx context.Context, clusterDeployment *kcmv1be
 		return fmt.Errorf("failed to create istio certificate: %v", err)
 	}
 
+	if !utils.IsClusterDeploymentReady(clusterDeployment) {
+		log.Info("Cluster deployment is not ready")
+		return nil
```

Comment:
This is confusing. 
As no error happen and the code invoked this TryCreate function doesn't see any difference where the clusterdeployment was created or not.